sdk/core/azure-core-test/src/main/java/com/azure/core/test/utils/TestProxyManager.java

```diff
@@ -45,6 +47,7 @@ public TestProxyManager(Path testClassPath) {
      */
     public void startProxy() {
         try {
+            semaphore.acquire();
```

Comment:
Use `synchronized` for this method as we expect this method to be called by single thread at a time.
```suggestion
    public synchronized void startProxy() {
        try {

```

Reply:
I'll rework this to be static so that `synchronized` will work.